src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -465,54 +472,90 @@ public LDAPSecurityRealm(String server, String rootDN, String userSearchBase, St
         this(server, rootDN, userSearchBase, userSearch, groupSearchBase, groupSearchFilter, groupMembershipStrategy, managerDN, managerPasswordSecret, inhibitInferRootDN, disableMailAddressResolver, cache, environmentProperties, displayNameAttributeName, mailAddressAttributeName, IdStrategy.CASE_INSENSITIVE, IdStrategy.CASE_INSENSITIVE);
     }
 
-    @DataBoundConstructor
+    /**
+     * @deprecated retained for backwards binary compatibility.
+     */
+    @Deprecated
     public LDAPSecurityRealm(String server, String rootDN, String userSearchBase, String userSearch, String groupSearchBase, String groupSearchFilter, LDAPGroupMembershipStrategy groupMembershipStrategy, String managerDN, Secret managerPasswordSecret, boolean inhibitInferRootDN, boolean disableMailAddressResolver, CacheConfiguration cache, EnvironmentProperty[] environmentProperties, String displayNameAttributeName, String mailAddressAttributeName, IdStrategy userIdStrategy, IdStrategy groupIdStrategy) {
-        this.server = server.trim();
-        this.managerDN = fixEmpty(managerDN);
-        this.managerPasswordSecret = managerPasswordSecret;
-        this.inhibitInferRootDN = inhibitInferRootDN;
-        if(!inhibitInferRootDN && fixEmptyAndTrim(rootDN)==null) rootDN= fixNull(inferRootDN(server));
-        this.rootDN = rootDN.trim();
-        this.userSearchBase = fixNull(userSearchBase).trim();
-        userSearch = fixEmptyAndTrim(userSearch);
-        this.userSearch = userSearch!=null ? userSearch : DescriptorImpl.DEFAULT_USER_SEARCH;
-        this.groupSearchBase = fixEmptyAndTrim(groupSearchBase);
-        this.groupSearchFilter = fixEmptyAndTrim(groupSearchFilter);
-        this.groupMembershipStrategy = groupMembershipStrategy == null ? new FromGroupSearchLDAPGroupMembershipStrategy("") : groupMembershipStrategy;
+        this(createLdapConfiguration(server, rootDN, userSearchBase, userSearch, groupSearchBase, groupSearchFilter, groupMembershipStrategy, managerDN, managerPasswordSecret, inhibitInferRootDN, environmentProperties, displayNameAttributeName, mailAddressAttributeName),
+                disableMailAddressResolver, cache, userIdStrategy, groupIdStrategy);
+    }
+
+    @DataBoundConstructor
+    public LDAPSecurityRealm(List<LDAPConfiguration> configurations, boolean disableMailAddressResolver, CacheConfiguration cache, IdStrategy userIdStrategy, IdStrategy groupIdStrategy) {
```

Comment:
Well I meant globally in Jenkins.
But one other reason to turn it off is to short cut the ldap lookup. I can't really do that for the multi ou since I need to find out what configuration the user is in before I can see if the configuration for that user allows resolving emails or not. Which means an ldap search to find out if we should not do any ldap searching :)